### PR TITLE
feat: Show "-" for null values in data cells 

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Show a placeholder '-' for null values in data cells.

Screenshot 2023-11-06 at 5 05 44 PM

## Test Plan

At the experiment list table, null values are denoted by '-' placeholder.
Sort and filter should work as expected. 

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1779

### CODE REVIEW

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx [hunk 1]
```diff
@@ -475,7 +475,22 @@ export const GlideTable: React.FC<GlideTableProps> = ({
...
-          return columnDefs[columnId]?.renderer?.(rowData, row) || loadingCell;
+          let cell = columnDefs[columnId]?.renderer?.(rowData, row);
+          switch (cell.kind) {
+            case GridCellKind.Text:
+            case GridCellKind.Number:
+              if (!cell.displayData || cell.displayData === '') {
+                cell = {
+                  ...cell,
+                  displayData: '-',
+                  themeOverride: { ...cell.themeOverride, textDark: 'LightGrey' },
```
Comment (gt2345) [inline webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:486]: Not really sure why, but using variable color such as `--theme-surface-on-weak` leads to weird inconsistent behavior

Comment (ashtonG) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:486]: could you try using `getCssVar` here to get the resolved variable value?

Comment (gt2345) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx]: That works! Thanks!
Also the `--theme-surface-on-weak` is just the default text color for the table

**Review by ashtonG — APPROVED** [review]
code looks good

**Pull request merged**